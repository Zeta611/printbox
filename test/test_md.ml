let b =
  let open PrintBox in
  tree (frame @@ text "root") [
    frame @@ text "child 1";
    text_with_style Style.preformatted "child 2";
    lines ["line 1"; "line 2"; "line 3"];
    frame @@ tree empty [
      tree (frame @@ text "header 3") [frame @@ text "subchild 3"]
    ];
    tree empty [
      tree (frame @@ text "header 4") [text "subchild 4"]
    ];
    frame @@ tree (text_with_style Style.preformatted "header 5")
      [lines_with_style Style.preformatted
        ["subchild 5"; "  body 5"; "end of 5"]];
    frame @@ text "child 6"
  ]

let () = print_endline "Test default:"

let () = print_endline @@ PrintBox_md.(to_string Config.default) b

let () = print_endline "Test foldable:"

let () = print_endline @@ PrintBox_md.(to_string Config.(foldable_trees default)) b

let () = print_endline "Test uniform:"

let () = print_endline @@ PrintBox_md.(to_string Config.(multiline_preformatted Code_quote uniform)) b

let () = print_endline "The end."
