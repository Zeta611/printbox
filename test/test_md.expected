Test default:
> root
- > child 1
- `child 2`
- line 1<br>
  line 2<br>
  line 3
- > 
  > - > header 3
  >   - > subchild 3
- 
  - > header 4
    - subchild 4
- > `header 5`
  > - ```
  >   subchild 5
  >     body 5
  >   end of 5
  >   ```
  >   
- > child 6

Test foldable:
<details><summary><span style="border:thin solid">root</span></summary>

- > child 1
- `child 2`
- line 1<br>
  line 2<br>
  line 3
- > <details><summary></summary>
  > 
  > - <details><summary><span style="border:thin solid">header 3</span></summary>
  >   
  >   - > subchild 3
  >   </details>
  > </details>
- <details><summary></summary>
  
  - <details><summary><span style="border:thin solid">header 4</span></summary>
    
    - subchild 4
    </details>
  </details>
- > <details><summary><span style="font-family: monospace">header 5</span></summary>
  > 
  > - ```
  >   subchild 5
  >     body 5
  >   end of 5
  >   ```
  >   
  > </details>
- > child 6
</details>

Test uniform:
<details><summary><span style="border:thin solid">root</span></summary>

- <span style="border:thin solid">child 1</span>
- <span style="font-family: monospace">child 2</span>
- line 1<br>
  line 2<br>
  line 3
- <span style="border:thin solid"><details><summary></summary>
  
  - <details><summary><span style="border:thin solid">header 3</span></summary>
    
    - <span style="border:thin solid">subchild 3</span>
    </details>
  </details></span>
- <details><summary></summary>
  
  - <details><summary><span style="border:thin solid">header 4</span></summary>
    
    - subchild 4
    </details>
  </details>
- <span style="border:thin solid"><details><summary><span style="font-family: monospace">header 5</span></summary>
  
  - `subchild 5`<br>
    `  body 5`<br>
    `end of 5`
  </details></span>
- <span style="border:thin solid">child 6</span>
</details>

The end.
